clusterous/cluster.py

```diff
@@ -1233,26 +1233,27 @@ def terminate_cluster(self, leave_shared_volume, force_delete_shared_volume):
             self._terminate_instances_and_wait(conn, instances)
 
         # Delete shared volume
-        volumes = conn.get_all_volumes(filters={'tag:Attached':self.cluster_name})
-        if volumes:
-            byo_volume = True if volumes else False
-            if byo_volume:
-                shared_volume = volumes[0]
-                shared_volume.remove_tags({'Attached': self.cluster_name})
-            else:
-                volumes = conn.get_all_volumes(filters={'tag:{0}'.format(defaults.instance_tag_key):self.cluster_name})
-                shared_volume = volumes[0]
-    
-            if leave_shared_volume:
-                self._logger.info('Shared volume "{0}" has not been deleted'.format(shared_volume.id))
-            else:
-                if force_delete_shared_volume or not byo_volume:
-                    if shared_volume.delete():
-                        self._logger.info('Shared volume "{0}" has been deleted'.format(shared_volume.id))
+        attached_volumes = conn.get_all_volumes(filters={'tag:Attached':self.cluster_name})
+        if attached_volumes:
+            for volume in attached_volumes:
```

Comment:
Any reason for this to be a loop? If anything, multiple shared volumes is an error condition.